pkg/logql/log/parser.go

```diff
@@ -339,10 +344,15 @@ func (l *LogfmtParser) Process(_ int64, line []byte, lbs *LabelsBuilder) ([]byte
 		}
 
 		val := l.dec.Value()
-		// the rune error replacement is rejected by Prometheus, so we skip it.
-		if bytes.ContainsRune(val, utf8.RuneError) {
-			val = nil
+
+		// the rune error replacement is rejected by Prometheus hence skiping them.
+		removeInvalidUtf := func(r rune) rune {
+			if r == utf8.RuneError {
+				return -1
+			}
+			return r
 		}
+		val = bytes.Map(removeInvalidUtf, val)
```

Comment:
see: https://github.com/grafana/loki/pull/14068#issuecomment-2337504002
```suggestion
		if bytes.ContainsRune(source, utf8.RuneError) {
			val = bytes.Map(removeInvalidUtf, val)
		}
```